File: client/src/app/modules/quotation/services/transfert-funds-type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { TransfertFundsType } from '../../quotation/model/TransfertFundsType';

@Injectable({
  providedIn: 'root'
})
export class TransfertFundsTypeService extends AppRestService<TransfertFundsType>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getTransfertFundsTypes() {
    return this.getListCached(new HttpParams(), "transfert-fund-types");
  }

  addOrUpdateTransfertFundsType(transfertFundsType: TransfertFundsType) {
    this.clearListCache(new HttpParams(), "transfert-fund-types");
    return this.addOrUpdate(new HttpParams(), "transfert-fund-type", transfertFundsType, "Enregistré", "Erreur lors de l'enregistrement");
  }
}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/TransfertFundsTypeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.TransfertFundsType;

public interface TransfertFundsTypeService {
    public List<TransfertFundsType> getTransfertFundsTypes();

    public TransfertFundsType getTransfertFundsType(Integer id);

    public TransfertFundsType addOrUpdateTransfertFundsType(TransfertFundsType transfertFundsType);
}
